kubectl-rook-ceph.sh

```diff
@@ -273,6 +276,94 @@ function run_purge_osd() {
     rook ceph osd remove --osd-ids=$1 --force-osd-removal=$force_removal"
 }
 
+####################################################################################################
+# 'kubectl rook-ceph debug commands
+####################################################################################################
+
+function update_deployment_spec_command(){
+  deployment_spec=$1
+  main_command='["sleep","infinity"]'
+  deployment_spec=$(jq .template.spec.containers[0].command="$main_command" <<< $deployment_spec)
+  deployment_spec=$(jq .template.spec.containers[0].args=[] <<< $deployment_spec)
+  echo "$deployment_spec" # for returning the updated deployment_spec
+}  
+
+function update_deployment_spec_image(){
+  deployment_spec=$1
+  alternalte_image=\"$2\"
+  deployment_spec=$(jq .template.spec.containers[0].image="$alternalte_image" <<< $deployment_spec)
+  deployment_spec=$(jq .template.spec.containers[0].env[0].value="$alternalte_image" <<< $deployment_spec)
+  echo "$deployment_spec" # for returning the updated deployment_spec
+}
+
+function verify_deployment_exist(){
+  if [[ $deployment_name != *"mon"* ]] && [[ $deployment_name != *"osd"* ]]; then
+    fail_error "only mon or osd deployment name can passed for starting debug mode"
+  fi
+  if output="$(KUBECTL_NS_CLUSTER get deployment "$deployment_name")"; then
+    echo "deployment $deployment_name exist, $output" 
+  else  
+    fail_error "deployment $deployment_name doesn't exist"
+  fi
+}
+
+function run_start_debug(){
+  [[ -z "${1:-""}" ]] && fail_error "Missing mon or osd deployment name"
+  deployment_name="$1"
+  alternate_image_flag="${2:-""}"
+  alternate_image="${3:-""}"
+  verify_deployment_exist "$deployment_name"
+  echo "setting debug mode for \"$deployment_name\""
+
+  # copy the deployment spec before scaling it down
+  deployment_spec=$(KUBECTL_NS_CLUSTER get  deployments  "$deployment_name" -o json | jq -r ".spec")
+  
+  # update the deployment_spec main container to be only a placeholder,
+  # also update image if alternate-image is passed
+  if [ -n "$alternate_image_flag" ]; then
+    if [ "$alternate_image_flag" == "--alternate-image" ]; then
+      if [ "$#" -eq 3 ]; then
+        echo "setting debug image to $alternate_image"
+        deployment_spec=$(update_deployment_spec_image "$deployment_spec" "$alternate_image")
+      else
+        fail_error "if --alternate-image flag is passed then image argument is mandatory"   
+      fi 
+    else
+      fail_error "--alternate-image should be passed as flag"
+    fi
+  fi  
+  echo "setting debug command to main container"
```

Comment:
```suggestion
  echo "setting debug command to main container"
```
can we get the name of `main/default` container?

Reply:
name should be simply mon or osd should we need to print that, by fetching it?